scalding-core/src/main/scala/com/twitter/scalding/Job.scala

```diff
@@ -251,6 +253,7 @@ class Job(val args: Args) extends FieldConversions with java.io.Serializable {
   // called before run
   // only override if you do not use flowDef
   def validate(): Unit = {
```

Comment:
this won't get called in the case where we write DOTs in Tool. Need to fix.

Reply:
:( i guess we lack a test for that path

Reply:
actually, I was wrong, we call `buildFlow`, which will make sure this stuff gets built, but we don't have a test for DOT generation, I don't think.

Reply:
Shame no test, but its not a critical path